packages/web/src/itinerary/plan.ts

```diff
@@ -0,0 +1,81 @@
+import Event from "./event";
+
+export default class Plan {
+    private _name: string;
+    private _events: Event[] = [];
+
+    constructor(name: string) {
+        this._name = name;
+    }
+
+    /**
+     * Gets the name of the plan
+     */
+    get name(): string {
+        return this._name;
+    }
+
+    /**
+     * Sets the name of the plan
+     */
+    set name(name: string) {
+        this._name = name;
+    }
+
+    /**
+     * Gets number of events in plan
+     */
+    get length(): number {
+        return this._events.length;
+    }
+
+    /**
+     * Gets all the events in a plan
+     */
+    get allEvents(): Event[] {
+        return this._events;
+    }
+
+    /**
+     * Adds a single event, returns true if successful
+     */
+    public addEvent(event: Event): boolean {
+        if (this._events.includes(event)) {
```

Comment:
We need to check if the given event id exists already, not the entire event since it could be a new event but just using the same id as a previous one.